ql/src/java/org/apache/hadoop/hive/ql/optimizer/calcite/rules/views/HiveRowIsDeletedPropagator.java

```diff
@@ -174,11 +174,11 @@ public RelNode visit(HiveJoin join) {
             .build();
   }
 
-  private static class InputRefShifter extends RexShuttle {
+  protected static class InputRefShifter extends RexShuttle {
```

Comment:
As it is now used from different classes, you may consider making it a top level class.